### PR TITLE
Add first version of Keycloak admin client based on Reactive REST Client

### LINKED ISSUE #20505: Cannot use rest-client-reactive-jackson together with quarkus-keycloak-admin-client
### Describe the bug

I am using quarkus-keycloak-admin-client in a project and want to add rest-client-reactive-jackson. However, I cannot use both libraries together because quarkus-keycloak-admin-client uses the old rest-client internally and I cannot have two rest clients in the same project.
When starting the project with `./mvnw quarkus:dev`, it complains about the conflict as follows:

```
Build step io.quarkus.deployment.steps.CapabilityAggregationStep#aggregateCapabilities threw an exception:
java.lang.IllegalStateException:
Please make sure there is only one provider of the following capabilities:
capability io.quarkus.rest.client is provided by:
  - io.quarkus:quarkus-rest-client-reactive::jar:2.2.3.Final
  - io.quarkus:quarkus-rest-client::jar:2.2.3.Final
```

### Expected behavior

I can use `rest-client-reactive-jackson` together with `quarkus-keycloak-admin-client` in a single project. Therefore, I would expect `quarkus-keycloak-admin-client` to recognize that there is a rest client already present in the project and use that (be it `quarkus-rest-client-reactive` or `quarkus-rest-client`).

### Actual behavior

When starting the application, I get the following exception:

```
Build step io.quarkus.deployment.steps.CapabilityAggregationStep#aggregateCapabilities threw an exception:
java.lang.IllegalStateException:
Please make sure there is only one provider of the following capabilities:
capability io.quarkus.rest.client is provided by:
  - io.quarkus:quarkus-rest-client-reactive::jar:2.2.3.Final
  - io.quarkus:quarkus-rest-client::jar:2.2.3.Final
```

### How to Reproduce?

1. Create a new project with: `mvn io.quarkus.platform:quarkus-maven-plugin:2.2.3.Final:create \
    -DprojectGroupId=org.acme \
    -DprojectArtifactId=rest-client-reactive-quickstart \
    -DclassName="org.acme.rest.client.CountriesResource" \
    -Dpath="/country" \
    -Dextensions="resteasy-reactive-jackson,rest-client-reactive-jackson,quarkus-keycloak-admin-client"`
2. `cd rest-client-reactive-quickstart`
3. `./mvnw quarkus:dev`

### Output of `uname -a` or `ver`

Darwin MacBook-Pro-KB.local 20.6.0 Darwin Kernel Version 20.6.0: Mon Aug 30 06:12:21 PDT 2021; root:xnu-7195.141.6~3/RELEASE_X86_64 x86_64

### Output of `java -version`

openjdk version "17" 2021-09-14 OpenJDK Runtime Environment Microsoft-26987 (build 17+35) OpenJDK 64-Bit Server VM Microsoft-26987 (build 17+35, mixed mode, sharing)

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.2.3.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.1

### Additional information

`./mvnw  dependency:tree`  reveals that Quarkus Keycloak Admin Client is using the rest client as a dependency:

```
[INFO] Scanning for projects...
[INFO] 
[INFO] --------------< org.acme:rest-client-reactive-quickstart >--------------
[INFO] Building rest-client-reactive-quickstart 1.0.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-dependency-plugin:2.8:tree (default-cli) @ rest-client-reactive-quickstart ---
[INFO] org.acme:rest-client-reactive-quickstart:jar:1.0.0-SNAPSHOT
[INFO] +- io.quarkus:quarkus-rest-client-reactive-jackson:jar:2.2.3.Final:compile
[INFO] |  +- io.quarkus:quarkus-resteasy-reactive-jackson-common:jar:2.2.3.Final:compile
[INFO] |  |  \- io.quarkus:quarkus-jackson:jar:2.2.3.Final:compile
[INFO] |  |     +- com.fasterxml.jackson.datatype:jackson-datatype-jsr310:jar:2.12.4:compile
[INFO] |  |     +- com.fasterxml.jackson.datatype:jackson-datatype-jdk8:jar:2.12.4:compile
[INFO] |  |     \- com.fasterxml.jackson.module:jackson-module-parameter-names:jar:2.12.4:compile
[INFO] |  \- io.quarkus:quarkus-rest-client-reactive:jar:2.2.3.Final:compile
[INFO] |     +- io.quarkus:quarkus-jaxrs-client-reactive:jar:2.2.3.Final:compile
[INFO] |     |  \- io.quarkus.resteasy.reactive:resteasy-reactive-client:jar:2.2.3.Final:compile
[INFO] |     |     \- io.vertx:vertx-web-client:jar:4.1.3:compile
[INFO] |     \- org.eclipse.microprofile.rest.client:microprofile-rest-client-api:jar:2.0:compile
[INFO] +- io.quarkus:quarkus-keycloak-admin-client:jar:2.2.3.Final:compile
[INFO] |  +- io.quarkus:quarkus-rest-client:jar:2.2.3.Final:compile
[INFO] |  |  +- io.quarkus:quarkus-resteasy-common:jar:2.2.3.Final:compile
[INFO] |  |  |  +- org.jboss.resteasy:resteasy-core:jar:4.7.0.Final:compile
[INFO] |  |  |  |  +- jakarta.validation:jakarta.validation-api:jar:2.0.2:compile
[INFO] |  |  |  |  \- com.ibm.async:asyncutil:jar:0.1.0:compile
[INFO] |  |  |  \- com.sun.activation:jakarta.activation:jar:1.2.1:compile
[INFO] |  |  +- io.quarkus:quarkus-apache-httpclient:jar:2.2.3.Final:compile
[INFO] |  |  +- org.jboss.resteasy:resteasy-client-microprofile:jar:4.7.0.Final:compile
[INFO] |  |  |  \- org.jboss.resteasy:resteasy-client-microprofile-base:jar:4.7.0.Final:compile
[INFO] |  |  |     \- org.jboss.resteasy:resteasy-cdi:jar:4.7.0.Final:compile
[INFO] |  |  \- jakarta.interceptor:jakarta.interceptor-api:jar:1.2.5:compile
[INFO] |  +- io.quarkus:quarkus-rest-client-jackson:jar:2.2.3.Final:compile
[INFO] |  |  +- org.jboss.resteasy:resteasy-jackson2-provider:jar:4.7.0.Final:compile
[INFO] |  |  |  +- com.fasterxml.jackson.core:jackson-annotations:jar:2.12.4:compile
[INFO] |  |  |  +- com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:jar:2.12.4:compile
[INFO] |  |  |  |  +- com.fasterxml.jackson.jaxrs:jackson-jaxrs-base:jar:2.12.4:compile
[INFO] |  |  |  |  \- com.fasterxml.jackson.module:jackson-module-jaxb-annotations:jar:2.12.4:compile
[INFO] |  |  |  \- com.github.java-json-tools:json-patch:jar:1.13:compile
[INFO] |  |  |     +- com.github.java-json-tools:msg-simple:jar:1.2:compile
[INFO] |  |  |     |  \- com.github.java-json-tools:btf:jar:1.3:compile
[INFO] |  |  |     \- com.github.java-json-tools:jackson-coreutils:jar:2.0:compile
[INFO] |  |  \- org.jboss.spec.javax.xml.bind:jboss-jaxb-api_2.3_spec:jar:2.0.0.Final:compile
[INFO] |  +- org.keycloak:keycloak-adapter-core:jar:14.0.0:compile
[INFO] |  +- org.keycloak:keycloak-core:jar:14.0.0:compile
[INFO] |  |  +- org.keycloak:keycloak-common:jar:14.0.0:compile
[INFO] |  |  +- org.bouncycastle:bcprov-jdk15on:jar:1.69:compile
[INFO] |  |  +- org.bouncycastle:bcpkix-jdk15on:jar:1.69:compile
[INFO] |  |  |  \- org.bouncycastle:bcutil-jdk15on:jar:1.69:compile
[INFO] |  |  +- com.fasterxml.jackson.core:jackson-core:jar:2.12.4:compile
[INFO] |  |  \- com.fasterxml.jackson.core:jackson-databind:jar:2.12.4:compile
[INFO] |  +- org.keycloak:keycloak-admin-client:jar:14.0.0:compile
[INFO] |  |  +- org.jboss.resteasy:resteasy-client:jar:4.7.0.Final:compile
[INFO] |  |  |  +- org.jboss.resteasy:resteasy-client-api:jar:4.7.0.Final:compile
[INFO] |  |  |  +- org.jboss.resteasy:resteasy-core-spi:jar:4.7.0.Final:compile
[INFO] |  |  |  +- commons-io:commons-io:jar:2.11.0:compile
[INFO] |  |  |  \- org.reactivestreams:reactive-streams:jar:1.0.3:compile
[INFO] |  |  +- org.jboss.resteasy:resteasy-multipart-provider:jar:4.7.0.Final:compile
[INFO] |  |  |  +- com.sun.mail:jakarta.mail:jar:1.6.5:compile
[INFO] |  |  |  +- org.apache.james:apache-mime4j-dom:jar:0.8.3:compile
[INFO] |  |  |  |  \- org.apache.james:apache-mime4j-core:jar:0.8.3:compile
[INFO] |  |  |  \- org.apache.james:apache-mime4j-storage:jar:0.8.3:compile
[INFO] |  |  \- org.jboss.resteasy:resteasy-jaxb-provider:jar:4.7.0.Final:compile
[INFO] |  |     \- org.glassfish.jaxb:jaxb-runtime:jar:2.3.3-b02:compile
[INFO] |  |        +- org.glassfish.jaxb:txw2:jar:2.3.3-b02:compile
[INFO] |  |        \- com.sun.istack:istack-commons-runtime:jar:3.0.10:compile
[INFO] |  +- org.apache.httpcomponents:httpasyncclient:jar:4.1.4:compile
[INFO] |  |  +- org.apache.httpcomponents:httpcore:jar:4.4.14:compile
[INFO] |  |  \- org.apache.httpcomponents:httpcore-nio:jar:4.4.14:compile
[INFO] |  +- org.keycloak:keycloak-authz-client:jar:14.0.0:compile
[INFO] |  |  \- org.jboss.logging:jboss-logging:jar:3.4.2.Final:compile
[INFO] |  \- org.jboss.logging:commons-logging-jboss-logging:jar:1.0.0.Final:compile
[INFO] +- io.quarkus:quarkus-resteasy-reactive-jackson:jar:2.2.3.Final:compile
[INFO] +- io.quarkus:quarkus-arc:jar:2.2.3.Final:compile
[INFO] |  +- io.quarkus.arc:arc:jar:2.2.3.Final:compile
[INFO] |  |  +- jakarta.enterprise:jakarta.enterprise.cdi-api:jar:2.0.2:compile
[INFO] |  |  |  \- jakarta.el:jakarta.el-api:jar:3.0.3:compile
[INFO] |  |  +- jakarta.annotation:jakarta.annotation-api:jar:1.3.5:compile
[INFO] |  |  \- jakarta.transaction:jakarta.transaction-api:jar:1.3.3:compile
[INFO] |  +- io.quarkus:quarkus-core:jar:2.2.3.Final:compile
[INFO] |  |  +- jakarta.inject:jakarta.inject-api:jar:1.0:compile
[INFO] |  |  +- io.quarkus:quarkus-ide-launcher:jar:2.2.3.Final:compile
[INFO] |  |  +- io.quarkus:quarkus-development-mode-spi:jar:2.2.3.Final:compile
[INFO] |  |  +- io.smallrye.config:smallrye-config:jar:2.4.4:compile
[INFO] |  |  |  \- io.smallrye.config:smallrye-config-core:jar:2.4.4:compile
[INFO] |  |  |     +- org.eclipse.microprofile.config:microprofile-config-api:jar:2.0:compile
[INFO] |  |  |     +- io.smallrye.common:smallrye-common-expression:jar:1.6.0:compile
[INFO] |  |  |     |  \- io.smallrye.common:smallrye-common-function:jar:1.6.0:compile
[INFO] |  |  |     +- io.smallrye.common:smallrye-common-constraint:jar:1.6.0:compile
[INFO] |  |  |     +- io.smallrye.common:smallrye-common-classloader:jar:1.6.0:compile
[INFO] |  |  |     \- io.smallrye.config:smallrye-config-common:jar:2.4.4:compile
[INFO] |  |  +- org.jboss.logmanager:jboss-logmanager-embedded:jar:1.0.9:compile
[INFO] |  |  +- org.jboss.logging:jboss-logging-annotations:jar:2.2.1.Final:compile
[INFO] |  |  +- org.jboss.threads:jboss-threads:jar:3.4.2.Final:compile
[INFO] |  |  +- org.slf4j:slf4j-api:jar:1.7.30:compile
[INFO] |  |  +- org.jboss.slf4j:slf4j-jboss-logmanager:jar:1.1.0.Final:compile
[INFO] |  |  +- org.graalvm.sdk:graal-sdk:jar:21.2.0:compile
[INFO] |  |  +- org.wildfly.common:wildfly-common:jar:1.5.4.Final-format-001:compile
[INFO] |  |  \- io.quarkus:quarkus-bootstrap-runner:jar:2.2.3.Final:compile
[INFO] |  \- org.eclipse.microprofile.context-propagation:microprofile-context-propagation-api:jar:1.2:compile
[INFO] +- io.quarkus:quarkus-resteasy-reactive:jar:2.2.3.Final:compile
[INFO] |  +- org.jboss.spec.javax.ws.rs:jboss-jaxrs-api_2.1_spec:jar:2.0.1.Final:compile
[INFO] |  +- io.quarkus:quarkus-resteasy-reactive-server-common:jar:2.2.3.Final:compile
[INFO] |  |  +- io.quarkus.resteasy.reactive:resteasy-reactive:jar:2.2.3.Final:compile
[INFO] |  |  \- io.quarkus.resteasy.reactive:resteasy-reactive-vertx:jar:2.2.3.Final:compile
[INFO] |  +- io.quarkus:quarkus-resteasy-reactive-common:jar:2.2.3.Final:compile
[INFO] |  |  +- io.quarkus.resteasy.reactive:resteasy-reactive-common:jar:2.2.3.Final:compile
[INFO] |  |  |  +- io.smallrye.reactive:mutiny:jar:1.0.0:compile
[INFO] |  |  |  \- io.smallrye.common:smallrye-common-annotation:jar:1.6.0:compile
[INFO] |  |  +- io.quarkus:quarkus-mutiny:jar:2.2.3.Final:compile
[INFO] |  |  |  +- io.quarkus:quarkus-smallrye-context-propagation:jar:2.2.3.Final:compile
[INFO] |  |  |  |  \- io.smallrye:smallrye-context-propagation:jar:1.2.0:compile
[INFO] |  |  |  |     +- io.smallrye:smallrye-context-propagation-api:jar:1.2.0:compile
[INFO] |  |  |  |     \- io.smallrye:smallrye-context-propagation-storage:jar:1.2.0:compile
[INFO] |  |  |  \- io.smallrye.reactive:mutiny-smallrye-context-propagation:jar:1.0.0:compile
[INFO] |  |  \- io.quarkus:quarkus-vertx-core:jar:2.2.3.Final:compile
[INFO] |  |     +- io.quarkus:quarkus-netty:jar:2.2.3.Final:compile
[INFO] |  |     |  +- io.netty:netty-codec:jar:4.1.67.Final:compile
[INFO] |  |     |  +- io.netty:netty-codec-http:jar:4.1.67.Final:compile
[INFO] |  |     |  +- io.netty:netty-codec-http2:jar:4.1.67.Final:compile
[INFO] |  |     |  \- io.netty:netty-handler:jar:4.1.67.Final:compile
[INFO] |  |     +- io.netty:netty-codec-haproxy:jar:4.1.67.Final:compile
[INFO] |  |     |  +- io.netty:netty-buffer:jar:4.1.67.Final:compile
[INFO] |  |     |  \- io.netty:netty-transport:jar:4.1.67.Final:compile
[INFO] |  |     \- io.vertx:vertx-core:jar:4.1.3:compile
[INFO] |  |        +- io.netty:netty-common:jar:4.1.67.Final:compile
[INFO] |  |        +- io.netty:netty-handler-proxy:jar:4.1.67.Final:compile
[INFO] |  |        |  \- io.netty:netty-codec-socks:jar:4.1.67.Final:compile
[INFO] |  |        +- io.netty:netty-resolver:jar:4.1.67.Final:compile
[INFO] |  |        \- io.netty:netty-resolver-dns:jar:4.1.67.Final:compile
[INFO] |  |           \- io.netty:netty-codec-dns:jar:4.1.67.Final:compile
[INFO] |  +- io.quarkus:quarkus-vertx-http:jar:2.2.3.Final:compile
[INFO] |  |  +- io.quarkus:quarkus-security-runtime-spi:jar:2.2.3.Final:compile
[INFO] |  |  +- io.quarkus:quarkus-vertx-http-dev-console-runtime-spi:jar:2.2.3.Final:compile
[INFO] |  |  +- io.quarkus.security:quarkus-security:jar:1.1.4.Final:compile
[INFO] |  |  \- io.vertx:vertx-web:jar:4.1.3:compile
[INFO] |  |     +- io.vertx:vertx-web-common:jar:4.1.3:compile
[INFO] |  |     +- io.vertx:vertx-auth-common:jar:4.1.3:compile
[INFO] |  |     \- io.vertx:vertx-bridge-common:jar:4.1.3:compile
[INFO] |  \- io.quarkus:quarkus-jsonp:jar:2.2.3.Final:compile
[INFO] |     \- org.glassfish:jakarta.json:jar:1.1.6:compile
[INFO] +- io.quarkus:quarkus-junit5:jar:2.2.3.Final:test
[INFO] |  +- io.quarkus:quarkus-bootstrap-core:jar:2.2.3.Final:test
[INFO] |  |  +- io.quarkus:quarkus-bootstrap-app-model:jar:2.2.3.Final:test
[INFO] |  |  +- io.quarkus:quarkus-bootstrap-maven-resolver:jar:2.2.3.Final:test
[INFO] |  |  |  +- org.apache.maven:maven-embedder:jar:3.8.1:test
[INFO] |  |  |  |  +- org.apache.maven:maven-settings:jar:3.8.1:test
[INFO] |  |  |  |  +- org.apache.maven:maven-core:jar:3.8.1:test
[INFO] |  |  |  |  |  +- org.apache.maven:maven-artifact:jar:3.8.1:test
[INFO] |  |  |  |  |  \- org.codehaus.plexus:plexus-component-annotations:jar:2.1.0:test
[INFO] |  |  |  |  +- org.apache.maven:maven-plugin-api:jar:3.8.1:test
[INFO] |  |  |  |  +- org.apache.maven:maven-model:jar:3.8.1:test
[INFO] |  |  |  |  +- org.apache.maven:maven-model-builder:jar:3.8.1:test
[INFO] |  |  |  |  +- org.apache.maven:maven-builder-support:jar:3.8.1:test
[INFO] |  |  |  |  +- org.apache.maven.resolver:maven-resolver-api:jar:1.6.2:test
[INFO] |  |  |  |  +- org.apache.maven.resolver:maven-resolver-util:jar:1.6.2:test
[INFO] |  |  |  |  +- org.apache.maven.shared:maven-shared-utils:jar:3.3.3:test
[INFO] |  |  |  |  +- com.google.inject:guice:jar:no_aop:4.2.1:test
[INFO] |  |  |  |  |  \- com.google.guava:guava:jar:30.1.1-jre:test
[INFO] |  |  |  |  |     +- com.google.guava:failureaccess:jar:1.0.1:test
[INFO] |  |  |  |  |     \- com.google.guava:listenablefuture:jar:9999.0-empty-to-avoid-conflict-with-guava:test
[INFO] |  |  |  |  +- org.codehaus.plexus:plexus-utils:jar:3.2.1:test
[INFO] |  |  |  |  +- org.codehaus.plexus:plexus-classworlds:jar:2.6.0:test
[INFO] |  |  |  |  \- commons-cli:commons-cli:jar:1.4:test
[INFO] |  |  |  +- org.eclipse.sisu:org.eclipse.sisu.plexus:jar:0.3.4:test
[INFO] |  |  |  +- org.apache.maven:maven-settings-builder:jar:3.8.1:test
[INFO] |  |  |  |  +- org.codehaus.plexus:plexus-interpolation:jar:1.25:test
[INFO] |  |  |  |  \- org.sonatype.plexus:plexus-sec-dispatcher:jar:1.4:test
[INFO] |  |  |  |     \- org.sonatype.plexus:plexus-cipher:jar:1.4:test
[INFO] |  |  |  +- org.apache.maven:maven-resolver-provider:jar:3.8.1:test
[INFO] |  |  |  |  +- org.apache.maven:maven-repository-metadata:jar:3.8.1:test
[INFO] |  |  |  |  +- org.apache.maven.resolver:maven-resolver-spi:jar:1.6.2:test
[INFO] |  |  |  |  \- org.apache.maven.resolver:maven-resolver-impl:jar:1.6.2:test
[INFO] |  |  |  +- org.apache.maven.resolver:maven-resolver-connector-basic:jar:1.6.2:test
[INFO] |  |  |  +- org.apache.maven.resolver:maven-resolver-transport-wagon:jar:1.6.2:test
[INFO] |  |  |  +- org.apache.maven.wagon:wagon-http:jar:3.4.3:test
[INFO] |  |  |  |  +- org.apache.maven.wagon:wagon-http-shared:jar:3.4.3:test
[INFO] |  |  |  |  |  \- org.jsoup:jsoup:jar:1.12.1:test
[INFO] |  |  |  |  \- org.apache.maven.wagon:wagon-provider-api:jar:3.4.3:test
[INFO] |  |  |  \- org.apache.maven.wagon:wagon-file:jar:3.4.3:test
[INFO] |  |  +- io.quarkus:quarkus-bootstrap-gradle-resolver:jar:2.2.3.Final:test
[INFO] |  |  \- io.smallrye.common:smallrye-common-io:jar:1.6.0:compile
[INFO] |  +- org.eclipse.sisu:org.eclipse.sisu.inject:jar:0.3.4:test
[INFO] |  +- io.quarkus:quarkus-test-common:jar:2.2.3.Final:test
[INFO] |  |  +- io.quarkus:quarkus-core-deployment:jar:2.2.3.Final:test
[INFO] |  |  |  +- org.aesh:readline:jar:2.1:test
[INFO] |  |  |  |  \- org.fusesource.jansi:jansi:jar:1.18:test
[INFO] |  |  |  +- io.quarkus.gizmo:gizmo:jar:1.0.9.Final:test
[INFO] |  |  |  |  \- org.ow2.asm:asm-util:jar:9.1:test
[INFO] |  |  |  |     +- org.ow2.asm:asm-tree:jar:9.1:test
[INFO] |  |  |  |     \- org.ow2.asm:asm-analysis:jar:9.1:test
[INFO] |  |  |  +- org.ow2.asm:asm:jar:9.1:test
[INFO] |  |  |  +- io.quarkus:quarkus-class-change-agent:jar:2.2.3.Final:test
[INFO] |  |  |  +- io.quarkus:quarkus-devtools-utilities:jar:2.2.3.Final:test
[INFO] |  |  |  +- io.quarkus:quarkus-builder:jar:2.2.3.Final:test
[INFO] |  |  |  \- org.junit.platform:junit-platform-launcher:jar:1.7.2:test
[INFO] |  |  +- io.quarkus:quarkus-jsonp-deployment:jar:2.2.3.Final:test
[INFO] |  |  \- org.jboss:jandex:jar:2.3.1.Final:test
[INFO] |  +- org.junit.jupiter:junit-jupiter:jar:5.7.2:test
[INFO] |  |  +- org.junit.jupiter:junit-jupiter-api:jar:5.7.2:test
[INFO] |  |  |  +- org.apiguardian:apiguardian-api:jar:1.1.0:test
[INFO] |  |  |  +- org.opentest4j:opentest4j:jar:1.2.0:test
[INFO] |  |  |  \- org.junit.platform:junit-platform-commons:jar:1.7.2:test
[INFO] |  |  +- org.junit.jupiter:junit-jupiter-params:jar:5.7.2:test
[INFO] |  |  \- org.junit.jupiter:junit-jupiter-engine:jar:5.7.2:test
[INFO] |  |     \- org.junit.platform:junit-platform-engine:jar:1.7.2:test
[INFO] |  \- com.thoughtworks.xstream:xstream:jar:1.4.17:test
[INFO] |     \- io.github.x-stream:mxparser:jar:1.2.1:test
[INFO] |        \- xmlpull:xmlpull:jar:1.1.3.1:test
[INFO] \- io.rest-assured:rest-assured:jar:4.4.0:test
[INFO]    +- org.codehaus.groovy:groovy:jar:3.0.8:test
[INFO]    +- org.codehaus.groovy:groovy-xml:jar:3.0.8:test
[INFO]    +- org.apache.httpcomponents:httpclient:jar:4.5.13:compile
[INFO]    |  \- commons-codec:commons-codec:jar:1.15:compile
[INFO]    +- org.apache.httpcomponents:httpmime:jar:4.5.13:test
[INFO]    +- org.hamcrest:hamcrest:jar:2.1:test
[INFO]    +- org.ccil.cowan.tagsoup:tagsoup:jar:1.2.1:test
[INFO]    +- io.rest-assured:json-path:jar:4.4.0:test
[INFO]    |  +- org.codehaus.groovy:groovy-json:jar:3.0.8:test
[INFO]    |  \- io.rest-assured:rest-assured-common:jar:4.4.0:test
[INFO]    \- io.rest-assured:xml-path:jar:4.4.0:test
[INFO]       +- org.apache.commons:commons-lang3:jar:3.12.0:test
[INFO]       +- jakarta.xml.bind:jakarta.xml.bind-api:jar:2.3.3:test
[INFO]       \- com.sun.xml.bind:jaxb-impl:jar:2.3.3:test
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  3.868 s
[INFO] Finished at: 2021-10-01T15:51:04+02:00
[INFO] ------------------------------------------------------------------------
```

### PR DESCRIPTION
**Pull request by geoand**
This is pretty much a straight copy of the existing Keycloak code
that removes various things that are not supported by Quarkus.

Furthermore, the base package was changed because there are cases
where the upstream Keycloak admin client still needs to be used (for example
in a QuarkusTestResourceLifecycleManager which runs before the Quarkus
application and therefore cannot use any Quarkus infrastructure. This case 
is exhibited by the second commit, which uses the upstream admin client for the 
`QuarkusTestResourceLifecycleManager`, but the new admin client for the application
itself).

Fixes: #20505

### CODE REVIEW

**Comment by geoand** [comment]
FYI, @sberyozkin, @cescoffier @michalszynkiewicz @pedroigor @gsmet

**Comment by sberyozkin** [comment]
@geoand Hi, that was super fast, thanks. I recall there was some discussion about keeping the actual Keycloak admin client source in the `independent-projects`, should it be placed there ? 
Right now it is a bit hard to see how we can try to start syncing up with at the SPI level

**Comment by geoand** [comment]
> I recall there was some discussion about keeping the actual Keycloak admin client source in the independent-projects, should it be placed there ?

We can certainly do that, I don't mind.
The reason I didn't do this is because I think that all this copied code will simply disappear when we update the upstream client, so we won't have to maintain any resources on our side - just an SPI implementation.

**Comment by sberyozkin** [comment]
Thanks @geoand, I believe @cescoffier was proposing it. I guess it can be easier for us to see some initial borders between the actual quarkus extension and the admin client source if this client source were located in `independent-projects`... I'd propose to move it there if there would be no objections

**Comment by geoand** [comment]
Let's see what @cescoffier says

**Comment by cescoffier** [comment]
I agree. 

I quickly looked at the pr and the naming is somewhat confusing (reactive but not a reactive api). So we would need to be smart on this.

Also, I believe we need to generate a diff with the upstream API to highlight the problem and see how the upstream needs to be adapted.

File: extensions/keycloak-admin-client-reactive/runtime/src/main/java/io/quarkus/keycloak/admin/client/reactive/Keycloak.java [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+        target = (WebTargetImpl) client.target(config.getServerUrl());
+        target.register(newAuthFilter());
+    }
+
+    // this code is much more complicated than expected because it needs to handle various permutations
+    // where beans may or may not exist
+    private ClientBuilderImpl registerJacksonProviders(ClientBuilderImpl clientBuilder) {
+        ArcContainer arcContainer = Arc.container();
+        if (arcContainer == null) {
```
Comment (michalszynkiewicz) [inline extensions/keycloak-admin-client-reactive/runtime/src/main/java/io/quarkus/keycloak/admin/client/reactive/Keycloak.java:70]: when can arc container be null?

Comment (geoand) [reply extensions/keycloak-admin-client-reactive/runtime/src/main/java/io/quarkus/keycloak/admin/client/reactive/Keycloak.java:70]: Initially I tried to make it work it tests environments as well, but that didn't pan out. So indeed this check could be removed.

**Comment by geoand** [comment]
Closing in favor of https://github.com/quarkusio/quarkus/pull/24300

**Pull request closed without merging**